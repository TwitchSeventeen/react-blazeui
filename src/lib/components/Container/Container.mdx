---
name: Container
route: /objects/container
menu: Objects
---

import { Playground, PropsTable } from 'docz'
import Container from './index';

# Container

Containers provide media query screen width restrictions on your content.

## Basic usage
```jsx
<Container size="xsmall"></Container>
<Container size="small"></Container>
<Container size="medium"></Container>
<Container size="large"></Container>
<Container size="xlarge"></Container>
<Container size="super"></Container>
```


## Properties
<PropsTable of={Container} />