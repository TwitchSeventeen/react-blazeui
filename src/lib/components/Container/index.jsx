import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Container(props) {
  const {
    children, className, size, dataName, as: Component, ...other
  } = props;
  const classes = classNames(
    'u-text',
    'o-container',
    `o-container--${size}`, className,
  );
  return (
    <Component
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </Component>
  );
}

Container.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * The container size
   */
  size: PropTypes.oneOf([
    'xsmall',
    'small',
    'medium',
    'large',
    'xlarge',
    'super',
  ]),
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
  /**
   * Element type to render as
   */
  as: PropTypes.elementType,
};

Container.defaultProps = {
  children: null,
  className: null,
  size: 'xlarge',
  dataName: 'container',
  as: 'div',
};


export default Container;
