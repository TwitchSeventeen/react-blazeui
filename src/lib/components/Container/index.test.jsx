import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Container from './index';

describe('Container', () => {
  it('should be defined', () => {
    expect(Container).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Container>
        lorem ipsum dolor
      </Container>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly a large container', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Container size="large">
        lorem ipsum dolor
        sit amet
      </Container>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
