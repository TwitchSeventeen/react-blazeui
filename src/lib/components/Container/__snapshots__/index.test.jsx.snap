// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<div
  className="o-container o-container--xlarge"
>
  lorem ipsum dolor
</div>
`;

exports[`Container renders correctly a large container 1`] = `
<div
  className="o-container o-container--large"
>
  lorem ipsum dolor sit amet
</div>
`;
